activation/nipost.go

```diff
@@ -211,21 +174,28 @@ func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.NIPos
 
 		submitCtx, cancel := context.WithDeadline(ctx, poetRoundStart)
 		defer cancel()
-		poetRequests, err := nb.submitPoetChallenges(submitCtx, poetProofDeadline, challengeHash.Bytes())
-		if err != nil {
+
+		if err := nb.submitPoetChallenges(submitCtx, poetProofDeadline, challenge.Hash().Bytes()); err != nil {
 			return nil, fmt.Errorf("submitting to poets: %w", err)
 		}
-		if len(poetRequests) == 0 {
-			return nil, &PoetSvcUnstableError{msg: "failed to submit challenge to any PoET", source: submitCtx.Err()}
+		count, err := nipost.PoetRegistrationCount(nb.localDB, nb.signer.NodeID())
+		if err != nil {
+			// TODO(mafa): panic instead?
```

Comment:
Opinions? Alternatively I could revert `nb.submitPoetChallenges` to return the list of registrations